mender-auth/ipc/server.cpp

```diff
@@ -70,93 +71,85 @@ auto NoOpHeaderHandler = [](http::ExpectedIncomingRequestPtr exp_req) {};
+		[this, &loop, &config, private_key_path, identity_script_path](
 			http::ExpectedIncomingRequestPtr exp_req) {
 			if (!exp_req) {
-				mlog::Error("Expected request failure: " + exp_req.error().message);
+				log::Error("Expected request failure: " + exp_req.error().message);
 				return;
 			}
 			auto req = exp_req.value();
 
 			auto exp_resp = req->MakeResponse();
 			if (!exp_resp) {
-				mlog::Error("Failed to create the response: " + exp_resp.error().message);
+				log::Error("Failed to create the response: " + exp_resp.error().message);
 				return;
 			}
 			auto resp = exp_resp.value();
 
-			mlog::Debug("Got path: " + req->GetPath());
+			log::Debug("Got path: " + req->GetPath());
 
 			if (req->GetPath() == "/getjwttoken") {
 				resp->SetHeader("X-MEN-JWTTOKEN", this->GetJWTToken());
 				resp->SetHeader("X-MEN-SERVERURL", this->GetServerURL());
 			} else if (req->GetPath() == "/fetchjwttoken") {
-				loop.Post([this, &config, &client, private_key_path, identity_script_path]() {
+				loop.Post([this, &config, private_key_path, identity_script_path]() {
 					auto err = auth_client::FetchJWTToken(
-						client,
+						this->client_,
 						config.server_url,
 						private_key_path,
 						identity_script_path,
-						std::bind(
-							&Caching::CacheAPIResponse,
-							this,
-							config.server_url,
-							std::placeholders::_1),
+						[this, &config](auth_client::APIResponse resp) {
+							this->CacheAPIResponse(config.server_url, resp);
+						},
 						config.tenant_token);
 					if (err != error::NoError) {
-						mlog::Error("Failed to set up the fetch request: " + err.message);
+						log::Error("Failed to set up the fetch request: " + err.message);
 					}
 				});
 			} else {
 				resp->SetStatusCodeAndMessage(404, "Not Found");
 				resp->AsyncReply([](error::Error err) {
 					if (err != error::NoError) {
-						mlog::Error("Failed to reply: " + err.message);
+						log::Error("Failed to reply: " + err.message);
 					};
 				});
 			}
 			resp->SetStatusCodeAndMessage(200, "Success");
 			resp->AsyncReply([](error::Error err) {
 				if (err != error::NoError) {
-					mlog::Error("Failed to reply: " + err.message);
+					log::Error("Failed to reply: " + err.message);
 				};
 			});
 		});
-	return err;
 }
 } // namespace webserver
 } // namespace type
 
-error::Error ListenForEvents(events::EventLoop &loop, const conf::MenderConfig &config) {
-	mlog::Trace("Setting up the webserver");
 
-	auto client_config {http::ClientConfig(config.server_certificate)};
-	http::Client client {client_config, loop};
+namespace server {
+error::Error Listen(events::EventLoop &loop, const conf::MenderConfig &config) {
+	log::Trace("Setting up the webserver");
 
 	const string server_url {"http://127.0.0.1:8001"};
-	http::ServerConfig server_config;
-	http::Server server(server_config, loop);
 
-	auto ipc_server {type::webserver::Caching {}};
+	type::webserver::Caching ipc_server {loop, config.server_certificate, config};
```

Comment:
I think there is still the same problem here: `Caching` is being constructed on the stack, and then the function returns, destroying it, and everything inside it. I suspect the whole API needs to be a class.

Reply:
lol, ur right of course. weak